$NetBSD: patch-ae,v 1.3 2001/09/23 10:53:38 frueauf Exp $

Since it is not fixed in 5.9.0, in fechtmail's mailing-list,
Sunil Shetye <shetye@bombay.retortsoft.com> proposed a patch :
in article <20010905130320.A26152@mcdonald.bombay.retortsoft.com>
which was provided in pr 14031 by Xavier HUMBERT <xavier@xavhome.fr.eu.org>.


--- sink.c-orig	Mon Aug  6 10:02:53 2001
+++ sink.c	Sun Sep 23 12:18:43 2001
@@ -586,6 +586,7 @@
 	const char	*ap;
 	char		options[MSGBUFSIZE]; 
 	char		addr[HOSTLEN+USERNAMELEN+1];
+	char		**from_responses;
 	int		total_addresses;
 
 	/*
@@ -665,6 +666,7 @@
 	total_addresses = 0;
 	for (idp = msg->recipients; idp; idp = idp->next)
 	    total_addresses++;
+	xalloca(from_responses, char **, sizeof(char *) * total_addresses);
 	for (idp = msg->recipients; idp; idp = idp->next)
 	    if (idp->val.status.mark == XMIT_ACCEPT)
 	    {
@@ -694,6 +696,20 @@
 
 		    handle_smtp_report(ctl, msg);
 
+#ifdef HAVE_SNPRINTF
+		    snprintf(errbuf, sizeof(errbuf), "%s: %s",
+			idp->id, smtp_response);
+#else
+		    strncpy(errbuf, idp->id, sizeof(errbuf));
+		    strcat(errbuf, ": ");
+		    strcat(errbuf, smtp_response);
+#endif /* HAVE_SNPRINTF */
+
+		    xalloca(from_responses[*bad_addresses],
+			char *,
+			strlen(errbuf)+1);
+		    strcpy(from_responses[*bad_addresses], errbuf);
+
 		    (*bad_addresses)++;
 		    idp->val.status.mark = XMIT_RCPTBAD;
 		    if (outlevel >= O_VERBOSE)
@@ -702,6 +718,10 @@
 			      ctl->listener, addr);
 		}
 	    }
+	if (*bad_addresses)
+	    send_bouncemail(ctl, msg, XMIT_RCPTBAD,
+		"Some addresses were rejected by the MDA fetchmail forwards to.\r\n",
+		*bad_addresses, from_responses);
 
 	/*
 	 * It's tempting to do local notification only if bouncemail was
